Sources/CasePaths/EnumReflection.swift

```diff
@@ -93,57 +110,240 @@ public func extract<Root, Value>(_ embed: @escaping (Value) -> Root) -> (Root) -
+    enumPtr.initialize(to: enumCase)
+
+    let untypedPtr = UnsafeMutableRawPointer(enumPtr)
+    valueWitnessTable.destructiveProjectEnumData(untypedPtr, ptr)
+
+    let valuesPtr = untypedPtr.assumingMemoryBound(to: Value.self)
+    let values = valuesPtr.pointee
+
+    valuesPtr.deinitialize(count: 1)
+    valuesPtr.deallocate()
+
+    return values
+  }
+
+  func indirectAssociatedValue<Enum, Value>(of enumCase: Enum, as type: Value.Type) -> Value {
+    // This is closely based on EnumImpl::subscript.
+    // https://github.com/apple/swift/blob/master/stdlib/public/runtime/ReflectionMirror.cpp
+
+    // I'll need access to this thing's bytes, which means it has to be a var.
+    var enumCase = enumCase
+
+    // Storage for an existential container with no conformances. In the C++ code, this is declared `Any enumCopy`, but `Any` is a `using` alias for `OpaqueExistentialContainer`.
+    var enumCopy = OpaqueExistentialContainer(nil, nil, nil, metadata: nil)
+
+    let answer: Value = withUnsafeMutablePointer(to: &enumCopy) { enumCopy in
+      // A “box” here means a value stored as the payload of a HeapObject.
+
+      let enumCopyContainer = allocateBoxForExistential(in: enumCopy)
+      initialize(enumCopyContainer, byCopying: &enumCase)
+
+      valueWitnessTable.destructiveProjectEnumData(enumCopyContainer, ptr)
+
+      // Note that this getter returns the pointer to the “full” metadata, which must be advanced by a word to become a normal metadata pointer.
+      let boxType = NativeObjectMetadata(
+        ptr: getBuiltinNativeObjectFullMetadata().advanced(by: pointerSize))
+      let pair = swift_allocBox(for: boxType.ptr)
+      boxType.initialize(pair.buffer, byTaking: enumCopyContainer)
+      deallocateBoxForExistential(in: enumCopy)
+
+      var valuePtr = pair.buffer
+      // The payload is always indirect in this method, so we have to open the box.
+      valuePtr = valuePtr.load(as: UnsafeMutableRawPointer.self)
+      valuePtr = swift_projectBox(valuePtr)
+
+      let value = valuePtr.load(as: type)
+      swift_release(pair.heapObject)
+      return value
+    }
+
+    return answer
+  }
+
+  func allocateBoxForExistential(in buffer: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer {
+    guard valueWitnessTable.flags.contains(.isNonInline) else {
+      return buffer
+    }
+
+    let (heapObject: heapObject, buffer: boxBuffer) = swift_allocBox(for: ptr)
+    buffer.storeBytes(of: heapObject, as: UnsafeMutableRawPointer.self)
+    return boxBuffer
+  }
+
+  func deallocateBoxForExistential(in buffer: UnsafeMutableRawPointer) {
+    guard valueWitnessTable.flags.contains(.isNonInline) else { return }
+    swift_deallocBox(buffer.load(as: UnsafeMutableRawPointer.self))
+  }
+}
+
+@_silgen_name("swift_allocBox")
+private func swift_allocBox(for metadata: UnsafeRawPointer) -> BoxPair
+
+@_silgen_name("swift_allocBox")
+private func swift_deallocBox(_ heapObject: UnsafeMutableRawPointer)
```

Comment:
Should the silgen name match here? 
```suggestion
@_silgen_name("swift_deallocBox")
private func swift_deallocBox(_ heapObject: UnsafeMutableRawPointer)
```

Reply:
lol yes